src/ApplicationConfigurator.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-cli for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-cli/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-cli/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Laminas\Cli;
+
+use Laminas\Cli\Listener\TerminateListener;
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Console\Application;
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @internal
+ */
+final class ApplicationConfigurator
```

Comment:
I really dislike this name - "Configurator" feels like a made-up Calvin and Hobbes term :rofl: .

Maybe `ApplicationProvisioner`?

Reply:
I dont have a strong need for this name so sure, can rename that.